src/index.rs

```diff
@@ -169,7 +169,29 @@ impl<T> BitcoinCoreRpcResultExt<T> for Result<T, bitcoincore_rpc::Error> {
   }
 }
 
-pub(crate) struct Index {
+/// An event that indicates that the location of an inscription has changed.
+/// Newly created inscriptions will include additional metadata including
+/// rarity, cursed status, charms, etc.
+#[derive(Debug, Clone)]
+pub struct LocationUpdateEvent {
```

Comment:
Let's make this an enum, with a variant for new inscriptions, and a variant for old inscriptions that have move:

```rust
enum Event {
  InscriptionCreated {
    id: InscriptionId,
    location: Option<SatPoint>,
    sequence_number: u32,
    block: u32,
    charms: u16,
    parent: Option<InscriptionId>,
  }
  InscriptionMoved {
    id: InscriptionId,
    old_location: SatPoint,
    new_location: SatPoint,
    block: u32,
    sequence_number: u32,
  }
}
```

- Location can be an option when created, and be `None` if unbound.
- When an inscription moves, both old_location and new_location should be set, since unbound inscriptions don't move.
- I like having charms just be a u16, since no charms is indicated by `0`.

Reply:
thanks for the suggestion, makes sense